fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1073,6 +1064,35 @@ public void rebuildIndex(@Nonnull FDBRecordStore store) {
         });
     }
 
+    private <T> CompletableFuture<T> runWithSynchronizedRunnerAndEndSession(
+            @Nonnull SynchronizedSessionRunner newSynchronizedRunner, @Nonnull Supplier<CompletableFuture<T>> runnable) {
+        if (this.synchronizedSessionRunner == null) {
+            this.synchronizedSessionRunner = newSynchronizedRunner;
+            return AsyncUtil.composeHandle(runnable.get(), (result, ex) -> {
+                if (this.synchronizedSessionRunner == newSynchronizedRunner) {
+                    this.synchronizedSessionRunner = null;
+                } else {
+                    LOGGER.warn(KeyValueLogMessage.of("synchronizedSessionRunner was modified during the run",
+                            LogMessageKeys.SESSION_ID, newSynchronizedRunner.getSessionId(),
+                            LogMessageKeys.INDEXER_SESSION_ID, this.synchronizedSessionRunner == null ? null : this.synchronizedSessionRunner.getSessionId()));
```

Comment:
There are similar NPE possibilities here as below (from races on the synchronized session runner object), though the window is smaller as there isn't a period where async work is shuffled off.

Reply:
Fixed.